app/test/vesting_test.go

```diff
@@ -0,0 +1,674 @@
+	// i.e. its start time yet to be reached.
+	vAcc, name, err := s.getAnUnusedPeriodicVestingAccount(block.Time.Unix() + 10)
+	assert.NoError(s.T(), err)
+	address := getAddress(name, s.cctx.Keyring).String()
+
+	balances, err := testfactory.GetAccountSpendableBalanceByBlock(s.cctx.GRPCClient, address, block)
+	assert.NoError(s.T(), err)
+
+	expectedSpendableBal := vAcc.GetVestedCoins(block.Time).AmountOf(app.BondDenom).Int64() + initBalanceForGasFee
+	assert.EqualValues(s.T(),
+		expectedSpendableBal,
+		balances.AmountOf(app.BondDenom).Int64(),
+		"spendable balance must match")
+}
+
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccountsDelegation() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// Find a periodic vesting account that that is currently in a vesting (locked) state
+	// i.e. its start time yet to be reached.
+	_, name, err := s.getAnUnusedPeriodicVestingAccount(tmtime.Now().Unix() + 10)
+	assert.NoError(s.T(), err)
+
+	s.testDelegatingVestingAmount(name)
+}
+
+// This test function tests delegation of a periodic vesting account that
+// has part of its allocation unlocked and part of it locked
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccountsDelegationPartiallyVested() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has some vesting (locked) and
+	// some vested (unlocked) balance
+	vAcc, name, err := s.getAnUnusedPeriodicVestingAccount(tmtime.Now().Unix() - 5)
+	assert.NoError(s.T(), err)
+
+	// Since we want a partially unlocked balance we need to wait until
+	// the first period has passed if not already
+	for vAcc.GetVestedCoins(tmtime.Now()).IsZero() {
+		time.Sleep(time.Second)
+	}
+	s.testDelegatingVestingAmount(name)
+}
+
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccountsClaimDelegationRewards() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has some vesting (locked) balance
+	// to be on the safe side we select one that starts unlocking in at least 20 seconds
+	_, name, err := s.getAnUnusedPeriodicVestingAccount(tmtime.Now().Unix() + 20)
+	assert.NoError(s.T(), err)
+
+	s.testDelegatingVestingAmount(name)
+	s.testClaimDelegationReward(name)
+}
+
+func (s *VestingModuleTestSuite) TestGenesisContinuousVestingAccountsSpendableBalanceLocked() {
+	block, err := s.cctx.LatestBlock()
+	assert.NoError(s.T(), err)
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find a continuous vesting account with locked balance
+	vAcc, name, err := s.getAnUnusedContinuousVestingAccount(block.Time.Unix() + 10)
+	assert.NoError(s.T(), err)
+	address := getAddress(name, s.cctx.Keyring).String()
+
+	balances, err := testfactory.GetAccountSpendableBalanceByBlock(s.cctx.GRPCClient, address, block)
+	assert.NoError(s.T(), err)
+
+	expectedSpendableBalAmount := vAcc.GetVestedCoins(block.Time).AmountOf(app.BondDenom).Int64() + initBalanceForGasFee
```

Comment:
[question] should this additionally verify that expectedSpendableBalAmount == initBalanceForGasFee or put another way that 0 coins have unlocked?

Reply:
Yes we can assert `vAcc.GetVestedCoins(block.Time).AmountOf(app.BondDenom).Int64()` is 0.